src/scripts/main.js

```diff
@@ -1,5 +1,32 @@
 'use strict';
+console.log(document);
+const spider = document.querySelector('.spider');
+const parent = spider.offsetParent;
```

Comment:
Similarly, ensure that `spider.offsetParent` is not `null`. If `spider` is not attached to the DOM or doesn't have an offset parent, this will cause a runtime error.